File: Solutions/217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashSet=set()
        
        for n in nums :
            if n in hashSet:
                return True
            hashSet.add(n)
        return False
      
    #Time complexity : O(N)
    #Space complexity : O(N)

File: Solutions/Java/217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for(int num : nums) {
            if(set.contains(num)) {
                return true;
            }
            set.add(num);
        }

        return false;
    }
}